medicines/search-client/src/lib.rs

```diff
@@ -166,43 +192,70 @@ impl CreateIndexEntry for AzureSearchClient {
 
 async fn search(
     search_term: &str,
+    result_count: Option<u32>,
+    offset: Option<u32>,
+    include_count: Option<bool>,
     client: &reqwest::Client,
     config: &AzureConfig,
-) -> Result<IndexResults, reqwest::Error> {
-    let req = build_search(search_term, &client, &config)?;
+) -> Result<IndexResults, anyhow::Error> {
+    let req = build_search(
+        search_term,
+        result_count,
+        offset,
+        include_count,
+        &client,
+        &config,
+    )?;
     tracing::debug!("Requesting from URL: {}", &req.url());
     client
         .execute(req)
         .await?
         .error_for_status()?
         .json::<IndexResults>()
         .await
+        .map_err(|e| anyhow::anyhow!(e))
 }
 
 fn build_search(
     search_term: &str,
+    result_count: Option<u32>,
+    offset: Option<u32>,
+    include_count: Option<bool>,
     client: &reqwest::Client,
     config: &AzureConfig,
-) -> Result<reqwest::Request, reqwest::Error> {
+) -> Result<reqwest::Request, anyhow::Error> {
     let base_url = format!(
         "https://{search_service}.search.windows.net/indexes/{search_index}/docs",
         search_service = config.search_service,
         search_index = config.search_index
     );
 
-    let req = client
+    let include_count = include_count.unwrap_or(false).to_string();
+
+    let request_builder = client
         .get(&base_url)
         .query(&[
             ("api-version", config.api_version.as_str()),
             ("highlight", "content"),
             ("queryType", "full"),
             ("search", search_term),
             ("scoringProfile", "preferKeywords"),
+            ("$count", &include_count),
         ])
-        .header("api-key", &config.api_key)
-        .build()?;
-
-    Ok(req)
+        .header("api-key", &config.api_key);
+
+    match (result_count, offset) {
+        (Some(result_count), Some(offset)) => Ok(request_builder
+            .query(&[
+                ("$top", result_count.to_string()),
+                ("$skip", offset.to_string()),
+            ])
+            .build()?),
+        (None, None) => Ok(request_builder.build()?),
+        _ => Err(anyhow::anyhow!(
+            "If one of result_count or offset is set, the other must be set as well."
```

Comment:
could they be unified into a `Pagination` type, then we pass in just `Option<Pagination>`? even `Option<(u32, u32>) should work